smart/model.py

```diff
@@ -2290,3 +2274,42 @@ def adjust_dt(self):
         #     dt_scale = min(dt_scale * 0.8, 0.8)
         dt_cur = float(self.dt) * dt_scale
         self.set_dt(dt_cur)
+
+    def create_restriction(self, u, mesh_function, value):
+        """
+        Restrict a function on a submesh to a subset of parent entities
+        (same dimension as the submesh)
+
+        :param u: Function on submesh
+        :param mesh_function: MeshFunction marking the subset of parent entities
+                            (same dimension as the cells of the submesh)
+        :param value: Value in MeshFunction marking the subset of parent entities
+        :return: New restricted function
+        """
+        submesh = u.function_space().mesh()
+
+        # Compute local cells in submesh marked by parent meshtag
+        mesh_ref = self.parent_mesh.dolfin_mesh
+        sub_to_parent_map = submesh.topology().mapping()[mesh_ref.id()].cell_map()
```

Comment:
Use the mesh ID from the stored parent mesh (mesh ID stored in the associated mesh function may be different I found).

Reply:
That is worrying, as then the cells could be mapped wrongly (it is not clear that there is a 1-1 correspondence between the parent mesh and whatever mesh the mesh function is associated with). 
we then need to look into how the meshfunction is created, as the assumption was that it was associated with the parent mesh.

Reply:
Yes, this could and likely should be fixed. I think the mesh and mesh function are reloaded when calling `mesh_tools.load_mesh` in the dendritic spine for example. It seems to not matter in this case, the mapping remains the same when I run it. But I guess we should define the extra mesh functions when the parent mesh is originally defined in the workflow.

Reply:
This is fixed now, extra mesh functions can (and should) be loaded in at the same time as the parent mesh.